glfw.lua

```diff
@@ -0,0 +1,147 @@
+-- Building libglfw for test consumption is done on an as-needed basis.
+-- In order to make this as easy as possible, RTC builds the library using the premake build system to utilize the
+-- advancements we've made in compilers and build scripts to simplify things without requiring many changes to the
+-- glfw project itself. For every platform, take a look at the configuration documentation for setup information.
+project "glfw"
+
+dofile(_BUILD_DIR .. "/shared_library.lua")
+
+configuration { "*" }
+
+uuid "11119b90-6c6e-4493-9860-86f7c2b581dc"
+
+defines {
+  "_GLFW_BUILD_DLL",
+}
+
+files {
+  "include/GLFW/glfw3.h",
+  "include/GLFW/glfw3native.h",
+  "src/internal.h",
+  "src/platform.h",
+  "src/mappings.h",
+  "src/context.c",
+  "src/init.c",
+  "src/input.c",
+  "src/monitor.c",
+  "src/platform.c",
+  "src/vulkan.c",
+  "src/window.c",
+  "src/egl_context.c",
+  "src/osmesa_context.c",
+  "src/null_platform.h",
+  "src/null_joystick.h",
+  "src/null_init.c",
+  "src/null_monitor.c",
+  "src/null_window.c",
+  "src/null_joystick.c",
+}
+
+includedirs {
+  "include/GLFW",
+}
+
+if (_PLATFORM_LINUX) then
+
```

Comment:
Can you add a comment here that xorg-dev needs to be installed to build?

Reply:
Updated. Plus bonus comment.